File: app/src/main/java/app/oswaldogh/plazashop/Ui/Products/Interface.java
package app.oswaldogh.plazashop.Ui.Products;

import java.util.ArrayList;

import app.oswaldogh.plazashop.Entities.Product;

/**
 * Created by oswaldogh89 on 04/01/18.
 */

public interface Interface {

    interface View {
        void createListProducts(ArrayList<Product> products);
        void addMoreProductsToList(ArrayList<Product> products);
        void hideLoadMoreProgress();
        void setProductListEmpty();
        boolean adapterIsNull();
    }

    interface Presenter {
        void onLoadProucts(ArrayList<Product> products);
    }

    interface Model {
        void getProducts(int totalProducts);
    }
}

File: app/src/main/java/app/oswaldogh/plazashop/Ui/Products/Interface.kt
package app.oswaldogh.plazashop.Ui.Products

import app.oswaldogh.plazashop.Entities.Product
import java.util.*

/**
 * Created by oswaldogomez on 05/12/17.
 */

interface Interface {
    interface View {
        fun showProducts(products: ArrayList<Product>)
    }

    interface Presenter {
        fun onLoadProucts(products: ArrayList<Product>)
    }

    interface Model {
        fun getProducts(totalProducts: Int)
    }
}
